v2/parser/parse.go

```diff
@@ -106,11 +111,14 @@ func (p *Parser) FindPackages(patterns ...string) ([]string, error) {
 		return results, nil
 	}
 
-	cfg := packages.Config{
-		Mode:       packages.NeedName | packages.NeedFiles,
-		BuildFlags: []string{"-tags", strings.Join(p.buildTags, ",")},
-		Tests:      false,
+	cfg := packages.Config{}
+	if baseCfg != nil { // This is to support tests, e.g. to inject a fake GOPATH or CWD.
+		cfg = *baseCfg
 	}
+	cfg.Mode |= packages.NeedName | packages.NeedFiles
+	cfg.BuildFlags = append(cfg.BuildFlags, "-tags", strings.Join(p.buildTags, ","))
+	cfg.Tests = false
```

Comment:
should this not be into an else statement then?
you re overriding the config passed

Reply:
Only overriding fields that we know we have to set, leaving other fields (like Dir and Env, which are what I need for the openapi tests to pass)

Reply:
I don't know how nitpicking is this as I´m not familiar with this codebase, but sounds that you could achieve the same just making `baseCfg *packages.Config` part of the Parser struct , since it seems you'll always need to access the private methods, and you´ll avoid to modify the methods

Reply:
I'd _really_ like to keep the override as constrained as possible until we have good reasons to expand and can reason about the interactions with the internal optionis we set... could keep the same signature (pass in a base config), but limit it to Dir and Env until we have reason to expand that, and only use Dir and Env here and in `loadPackages`, e.g.:

```diff
+if baseCfg == nil {
+ 		baseCfg = &packages.Config{}
+}
 cfg := packages.Config{
 		Mode:       packages.NeedName | packages.NeedFiles,
 		BuildFlags: []string{"-tags", strings.Join(p.buildTags, ",")},
 		Tests:      false,
+		Dir:        baseCfg.Dir,
+		Env:        baseCfg.Env, 
 }
```

Reply:
That works, too